examples/fit/README.md

```diff
@@ -0,0 +1,69 @@
+# Flexible Vision Transformer for Diffusion Model (FiT)
+
+*Note: This project is still work in progress.*
+
+This folder contains [FiT](https://arxiv.org/abs/2402.12376) models implemented with [MindSpore](https://www.mindspore.cn/),
+
+Nature is infinitely resolution-free. In the context of this reality, existing diffusion models, such as Diffusion Transformers, often face challenges when processing image resolutions outside of their trained domain. To overcome this limitation, the Flexible Vision Transformer (FiT), a transformer architecture is specifically designed for generating images with unrestricted resolutions and aspect ratios. Unlike traditional methods that perceive images as static-resolution grids, FiT conceptualizes images as sequences of dynamically-sized tokens. This perspective enables a flexible training strategy that effortlessly adapts to diverse aspect ratios during both training and inference phases, thus promoting resolution generalization and eliminating biases induced by image cropping. Enhanced by a meticulously adjusted network structure and the integration of training-free extrapolation techniques, FiT exhibits remarkable flexibility in resolution extrapolation generation. Comprehensive experiments demonstrate the exceptional performance of FiT across a broad range of resolutions, showcasing its effectiveness both within and beyond its training resolution distribution.[1]
+
+## Get Started
+
+### Pretrained Checkpoints
+
+## Sampling
+
+To run inference of `FiT-XL/2` model with the `256x256` image size on Ascend devices, you can use:
+```bash
+python sample.py --imagegrid True
+
+```
+You can also adjust the image size by add the flag `--image_height` and `--image_width`. For example, you can run
+```bash
+python sample.py --imagegrid True --image_height 320 --image_width 160
+```
+to generate image with 160x320 size.
+
+## Training with ImageNet format
```

Comment:
Before introducing the training scripts for ImageNet dataset, please also introduce where to download ImageNet dataset, and the folder structure of the downloaded data.

Reply:
Done.